src/DelimitedDataParser/DelimitedDataReader.cs

```diff
@@ -253,7 +253,7 @@ public override long GetChars(int ordinal, long dataOffset, char[] buffer, int b
 
         public override string GetDataTypeName(int ordinal)
         {
-            if (ordinal < 0 || ordinal > _currentRow.Count - 1)
+            if (!_yieldExistingRow && (ordinal < 0 || ordinal > _currentRow.Count - 1))
```

Comment:
This change appears to be disabling argument range checking if we're due to "yield an existing row". Should we not instead be modifying the checks that we perform depending on whether `_yieldExistingRow` is set? For example, if this is `false`, then we check against `_fieldNameLookup.Count`, otherwise we check against `_currentRow.Count`?